create.ps1

```diff
@@ -1,258 +1,449 @@
-            $UserResponse = Invoke-RestMethod @LisaRequest @SplatParams
-            $outputContext.AccountReference = $UserResponse.objectId
-        }
-        else {
-            Write-Verbose -Verbose ($Body | ConvertTo-Json)
-            $outputContext.AccountReference = 'unknown'
-        }
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = "Created account with displayName [$($createAccountBody.displayName)] and userPrincipalName [$($createAccountBody.userPrincipalName)] with AccountReference: $($outputContext.AccountReference | ConvertTo-Json)."
+                        IsError = $false
+                    })
+            }
+            else {
+                Write-Warning "DryRun: Would create account with displayName [$($createAccountBody.displayName)] and userPrincipalName [$($createAccountBody.userPrincipalName)]."
+            }
+            #endregion Create account
 
+            #region Update account
+            # API docs: https://mwpapi.kpnwerkplek.com/index.html, specific API call: PATCH /api/users/{identifier}/bulk
+            $actionMessage = "updating created account with AccountReference: $($outputContext.AccountReference | ConvertTo-Json)"
 
-        $outputContext.Data | Add-Member -NotePropertyMembers @{
-            password = $UserResponse.temporaryPassword
-        } -Force
+            # Update account with all other fields than the required fields
+            $updateAccountBody = [PSCustomObject]@{}
+            foreach ($accountProperty in $account.PsObject.Properties | Where-Object { $null -ne $_.Value -and $_.Name -notin $requiredFields }) {
+                $updateAccountBody | Add-Member -MemberType NoteProperty -Name $accountProperty.Name -Value $accountProperty.Value -Force
+            }
 
-        #Update the user with all other props
-        $Body = $outputContext.Data | Select-Object -Property *, "accountEnabled" -ExcludeProperty @(
-            $CreationProperties, "password"
-        )
+            $updateAccountSplatParams = @{
+                Uri         = "$($actionContext.Configuration.MWPApiBaseUrl)/users/$($outputContext.AccountReference)/bulk"
+                Method      = "PATCH"
+                Body        = ($updateAccountBody | ConvertTo-Json -Depth 10)
+                ContentType = 'application/json; charset=utf-8'
+                Verbose     = $false
+                ErrorAction = "Stop"
+            }
 
-        # Force the account disabled
-        $Body.accountEnabled = $False
+            Write-Verbose "SplatParams: $($updateAccountSplatParams | ConvertTo-Json)"
 
-        $SplatParams = @{
-            Uri    = "$($actionContext.Configuration.BaseUrl)/Users/$($outputContext.AccountReference)/bulk"
-            Method = "Patch"
-            Body   = $Body
-        }
+            if (-Not($actionContext.DryRun -eq $true)) {
+                # Add header after printing splat
```

Comment:
Dit soort comments mogen weg.

Reply:
Ik snap je punt, maar het staat ook niet in de weg of werkt er minder door, dus ik laat het voor nu zo.